internal/testutil/mig/test_case_test.go

```diff
@@ -0,0 +1,115 @@
+package mig_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/mig"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConvertToMigration(t *testing.T) {
+	acc.SkipInUnitTest(t) // requires MONGODB_ATLAS_LAST_VERSION
```

Comment:
is there any way of modifying the os environment to set the MONGODB_ATLAS_LAST_VERSION but only for this test?
Would like to run this as part of the unit tests.

Reply:
I believe we slightly discussed this, but could you add the logic for checking that this env variable is defined within the PreCheck function, and not as part of the flow for converting one TestCase to another?
This way we can unit testing the convert function without having to worry about the env variable

Reply:
I believe we slightly discussed this, but could you add the logic for checking that this env variable is defined within the PreCheck function, and not as part of the flow for converting one TestCase to another?
This way we can unit testing the convert function without having to worry about the env variable

Reply:
Two problems:
1. `ExternalProviders()` is reading that variable
2. The unit test also wants to check that preCheck have been called

Python has a way of monkey-patching the env_var for a test. Is this possible?

Reply:
It existed! 🥳 
Checkout: 8e969ad57cf6ff3dc5ed16f21501376469921b46